src/content/insertable-streams/video-recording/js/main.js

```diff
@@ -0,0 +1,154 @@
+/*
+ *  Copyright (c) 2021 The WebRTC project authors. All Rights Reserved.
+ *
+ *  Use of this source code is governed by a BSD-style license
+ *  that can be found in the LICENSE file in the root of the source
+ *  tree.
+ */
+
+'use strict';
+
+/* global MediaStreamTrackProcessor, MediaStreamTrackGenerator */
+if (typeof MediaStreamTrackProcessor === 'undefined' ||
+  typeof MediaStreamTrackGenerator === 'undefined') {
+  alert(
+    'Your browser does not support the experimental MediaStreamTrack API ' +
+    'for Insertable Streams of Media. See the note at the bottom of the ' +
+    'page.');
+}
+
+const startButton = document.getElementById('startButton');
+const slowDownButton = document.getElementById('slowDownButton');
+const stopButton = document.getElementById('stopButton');
+const originalVideo = document.getElementById('originalVideo');
+const recordedOriginalVideo = document.getElementById('recordedOriginalVideo');
+const mirroredWithCanvasVideo = document.getElementById('mirroredWithCanvasVideo');
+const recordedMirroredWithCanvasVideo = document.getElementById('recordedMirroredWithCanvasVideo');
+const mirroredInWebWorkerVideo = document.getElementById('mirroredInWebWorkerVideo');
+const recordedMirroredInWorkerVideo = document.getElementById('recordedMirroredInWorkerVideo');
+
+const worker = new Worker('./js/worker.js', { name: 'Crop worker' });
+
+
+class VideoRecorder {
+  constructor(stream, outputVideoElement) {
+    this.videoElement = outputVideoElement;
+    this.mediaRecorder = new MediaRecorder(stream);
+
+    this.mediaRecorder.ondataavailable = (event) => {
+      if (event.data.size > 0) {
+        this.recordedBlob.push(event.data);
+      }
+    };
+
+    this.mediaRecorder.onerror = (e) => {
+      throw e;
+    };
+
+    this.recordedBlob = [];
+  }
+
+  start() {
+    this.mediaRecorder.start(1000);
+  }
+
+  stop() {
+    this.mediaRecorder.stop();
+    console.log('stopped');
+    const blob = new Blob(this.recordedBlob, { type: 'video/webm' });
+    const url = URL.createObjectURL(blob);
+    this.videoElement.src = url;
+  }
+}
+
+let recorders = [];
+
+startButton.addEventListener('click', async () => {
+  const stream = await navigator.mediaDevices.getUserMedia({ video: { width: 1280, height: 720 } });
```

Comment:
Hard coded camera size will fail on many device.
Use video: true constraints and videoWidth/Height on video tag once metadataloaded to get size instead or VideoTrack width or height settings.